dkg-mock-blockchain/src/data_types.rs

```diff
@@ -0,0 +1,212 @@
+use crate::{mock_blockchain_config::ErrorCase, FinalityNotification, ImportNotification};
+use codec::{Decode, Encode};
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+#[derive(Debug, Clone)]
+pub enum MockBlockChainEvent<B: crate::BlockTraitForTest> {
+	FinalityNotification { notification: FinalityNotification<B> },
+	ImportNotification { notification: ImportNotification<B> },
+	TestCase { trace_id: Uuid, test: TestCase },
+}
+
+pub trait BlockTraitForTest: sp_runtime::traits::Block + Unpin {}
+impl<T: sp_runtime::traits::Block + Unpin> BlockTraitForTest for T {}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct AttachedCommandMetadata {
+	// Specifies the command which the receiving MockClient is expected to take when receiving this
+	// message. The message may purposefully attempt to cause an error to try to see how receiving
+	// nodes react to the adversity
+	command: AttachedCommand,
+	// trace ID
+	trace_id: Uuid,
+}
+
+/// A command given to the receiving MockClient
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub enum AttachedCommand {
+	// Tells the client to process the requests as normal
+	ProcessAsNormal,
+	// Tells the client to not process the request (for this round)
+	ErrorDontProcessRequest,
+}
+
+/// When a MockClient receives a message, it should attempt to send information
+/// about its internal state back to the MockBlockchain server for centralized
+/// introspection
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct MockClientResponse {
+	pub error: Option<String>,
+	pub success: bool,
```

Comment:
This could be just Result, or make `error: Option<String>` where a `success` is a method that just return `self.error.is_none()`